Framework/MDAlgorithms/src/IntegrateQLabEvents.cpp

```diff
@@ -301,6 +316,27 @@ PeakShapeEllipsoid_const_sptr IntegrateQLabEvents::ellipseIntegrateEvents(
   inti = peak_w_back.first - ratio * backgrd.first;
   sigi = sqrt(peak_w_back.second + ratio * ratio * backgrd.second);
 
+  if (inti < 0) {
+    std::ostringstream msg;
+    msg << "Negative intensity found: " << inti << "\n"
+        << "Please use slice viewer to check the peak with negative intensity to decide:\n"
+        << "-- adjust peak and background raidus\n"
+        << "-- prune false positive indexation results\n";
+    g_log.notice() << msg.str();
+    // debug message
+    std::ostringstream debugmsg;
+    debugmsg << "peak_radius = " << peak_radius << "\n"
+             << "back_inner_radius = " << back_inner_radius << "\n"
+             << "back_outer_radius = " << back_outer_radius << "\n"
+             << "sigmas = (" << sigmas[0] << "," << sigmas[1] << "," << sigmas[2] << ")\n"
+             << "r1, r2, r3 = " << r1 << "," << r2 << "," << r3 << "\n"
+             << "peak_w_back.first = " << peak_w_back.first << "\n"
+             << "backgrd.first = " << backgrd.first << "\n"
+             << "ratio = " << ratio << "\n"
+             << "inti = peak_w_back.first - ratio * backgrd.first = " << inti << "\n";
+    g_log.debug() << debugmsg.str();
+  }
+
```

Comment:
Could you rebase the branch to current ornl-next?  I don't think the changes here is relevant to this PR.

Reply:
I just did. The changes from IntegrateQLabEvents.cpp are gone. You will have to give your approval again, though

Reply:
already did :sunglasses: